server/src/main/java/org/elasticsearch/cluster/routing/allocation/allocator/BalancedShardsAllocator.java

```diff
@@ -1172,12 +1167,12 @@ public boolean containsShard(ShardRouting shard) {
     }
 
     static final class ModelIndex implements Iterable<ShardRouting> {
-        private final String id;
+        private final String indexName;
```

Comment:
Is this field used for anything?

Reply:
It was used here:

```
                indices.put(index.getIndexId(), index);
```

That wasn't necessary tho, it's the same as `shard.getIndexName()` which was already in scope. I imagine it was useful for development/debugging in the past but no need for it any more IMO.